examples/brainfuck.rs

```diff
@@ -32,9 +32,10 @@ fn parser() -> impl Parser<char, Vec<Instr>, Error = Simple<char>> {
             just('.').to(Write),
         ))
         .or(bf.delimited_by(just('['), just(']')).map(Loop))
-        .recover_with(nested_delimiters('[', ']', [], |_| Invalid))
-        .recover_with(skip_then_retry_until([']']))
+        .recover_with(any().repeated().delimited_by(just('['), just(']')).to(Invalid))
```

Comment:
It would be nice to have a few built-in recovery parsers akin to `nested_delimiters` and others.

Fwiw, this code is not equivalent to `nested_delimiters` because it doesn't account for nesting. For example, this will happily accept `[...[...]` as a recovery pattern, even though it should still be looking for a final `]`.

Reply:
I knew it wasn't identical, but I was largely just getting things working. And I agree, lots of things already use `nested_delimiters` and it's probably useful to have some builtin strategies, even if they're no longer separate from other parsers. I'll see if I can port at least `nested_delimiters` and maybe some of the others, put them in a `recovery` module for now unless you think there's a better place.